code/modules/admin/permissionedit.dm

```diff
@@ -165,7 +165,10 @@
 			return
 	switch(task)
 		if("add")
-			admin_ckey = add_admin(null, use_db)
```

Comment:
This should be functionally identical to `admin_ckey = add_admin(admin_ckey, use_db)` so the `if()` below isn't needed.

Reply:
if you pass null instead of admin_ckey, it asks you to input the ckey of the person instead of using the ckey that we're already trying to promote

Reply:
I now realize that admin_ckey will pass as null if it's not set, i'm a moron.